barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -70,17 +72,28 @@ namespace bb {
  documentation </a>
  */
 template <typename Curve> class IPA {
+    // clang-fromat on
     using Fr = typename Curve::ScalarField;
     using GroupElement = typename Curve::Element;
     using Commitment = typename Curve::AffineElement;
     using CK = CommitmentKey<Curve>;
     using VK = VerifierCommitmentKey<Curve>;
     using Polynomial = bb::Polynomial<Fr>;
 
-  public:
+// These allow access to internal functions so that we can never use a mock transcript unless it's fuzzing or testing of IPA specifically
+#ifdef IPA_TEST
+    FRIEND_TEST(IPATest, ChallengesAreZero);
+    FRIEND_TEST(IPATest, AIsZeroAfterOneRound);
+#endif
+#ifdef IPA_FUZZ_TEST
+    friend class ProxyCaller;
+#endif
+    // clang-format off
+
     /**
-     * @brief Compute an inner product argument proof for opening a single polynomial at a single evaluation point
+     * @brief Compute an inner product argument proof for opening a single polynomial at a single evaluation point.
      *
+     * @tparam Transcript Transcript type. Useful for testing
```

Comment:
typo

Reply:
Where exactly?

Reply:
Transcript Transcript?

Reply:
One is an automatic name of the parameter, the other is "Transcript type"